### DISCUSSION TITLE
CloseButton Component will not work in next.js when imported from @headlessui/react

### DISCUSSION THREAD

**Discussion by Elieserlaguerre** [tailwindlabs/headlessui#3651] [Help]
**What package within Headless UI are you using?**
I'm using the Disclosure from @headlessui/react

**What version of that package are you using?**
I'm using version:  "^2.2.0"

**What browser are you using?**
 Chrome

**Reproduction URL**
I don't know how to create/provide a reproduction URL. 

**Describe your issue**

Describe the problem you're seeing, any important steps to reproduce, and what behavior you expect instead.
When I click on the CloseButton component, it does nothing except redirect the client to another page, which is what I want. However, it does not close the DisclosurePanel, which is what I want.

**Comment by RobinMalfait** [comment]
Hey!

Make sure that the `CloseButton` is inside of a `DisclosurePanel`. If the `CloseButton` is used outside of a closeable component then it will be a no-op.

Can you share a minimal reproduction repo so we can take a look?
